dom.bs

```diff
@@ -1197,11 +1205,12 @@ for discussion).
      <var>parent</var>.
 
      <li><p><a>Append to an event path</a> with <var>event</var>, <var>parent</var>, null,
-     <var>relatedTarget</var>, and <var>slot-in-closed-tree</var>.
+     <var>relatedTargets</var>, and <var>slot-in-closed-tree</var>.
     </ol>
 
-   <li><p>Otherwise, if <var>parent</var> and <var>relatedTarget</var> are identical, then set
-   <var>parent</var> to null.
+   <li><p>Otherwise, if <var>relatedTargets</var> <a for=list>contains</a> <var>parent</var>, then
+   set <var>parent</var> to null.
```

Comment:
Gut feeling is that also this has similar problem.

Reply:
Yup, I confirmed the Blink's implementation.
We need to distinguish *relatedTarget* and *relatedTargets* somehow because only *relatedTarget* (in the sense of before this PR) should be considered here.
"*relaterTagets* contains parent" causes a sort of *false positive*.

Reply:
How should touch targets be adjusted though?

Reply:
Since the feedback I got before was that whenever we have relatedTarget handling, we also need to consider touch target handling.

Reply:
These steps always need some time to understand its intention exactly. :(

My intention is:
- All touch targets should be adjusted in the same way as relatedTarget should be adjusted. They both should be adjusted.
- However, there is one difference between relatedTarget and touch targets. Only relatedTarget should be used to decide to exit this while loop "11. While parent is non-null". This step sets *parent* to null, thus I think this is an attempt to exit the loop.
To decide to exit this loop or not, in other words, to decide whether an event should be stopped at a shadow root or not,  only relatedTarget should be used as its judgement. Touch targets shouldn't be used there.

Since TouchEvent doesn't look to have a relatedTarget to me, TouchEvent never set parent to null here. In other words, TouchEvent is never stopped at a shadow root even if its target and one of touch targets are same.

Reply:
Okay, so we effectively need to have relatedTarget as an internal slot and touchTargets as an internal slot and most of the time we do the same thing for both, but not there and not at the other place @smaug---- pointed out (which is hidden now for some reason).

I can make that change. I'd love some help with tests for touch targets btw since I don't really have a good way to run them here.

Reply:
(The alternative option is that we keep one slot but branch on event interface, but that seems a little icky, but let me know if you'd prefer that.)

Reply:
I would love to help to write more tests, though I'm afraid I can write tests only for synthetic TouchEvents. I am not sure how to write a test for non-synthetic TouchEvents.

> (The alternative option is that we keep one slot but branch on event interface, but that seems a little icky, but let me know if you'd prefer that.)

I don't have any preference.

btw, I am wondering how these *relatedTargets* will be used in TouchEvent spec.
TouchEvents has 3 *relatedTargets* lists.

```
    readonly        attribute TouchList touches;
    readonly        attribute TouchList targetTouches;
    readonly        attribute TouchList changedTouches;
```

Reply:
I was thinking that since those lists are all immutable, a flattened list backing them is fine, but that only works if retargeting is consistent regardless of which list.

Reply:
I see. Your assumption is correct. Retargeting is consistent regardless of which list.